tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/OptimizerCore.scala

```diff
@@ -2168,6 +2168,18 @@ private[optimizer] abstract class OptimizerCore(
     }
   }
 
+  private def literalToString(l: Literal): String = l match {
+    case StringLiteral(str) => str
+    case FloatLiteral(flt) => flt.toDouble.toString
```

Comment:
This also needs `isValidInt` something, otherwise `1.0f.toString` will badly constant-fold.